### PR TITLE
[DUOS-1740][risk=no] Update tests for JUnit 5 and Dropwizard 2

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1740

Significant refactor of tests.
* Removal of deprecated DW test classes
* Update most tests to JUnit 5
* Did not touch the parameterized tests, those are essentially running under JUnit 4

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package org.broadinstitute.dsde.consent.ontology.datause;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.mindswap.pellet.utils.ATermUtils.assertTrue;
```
Comment (connorlbark) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java]: Is this intentionally not being imported from junit?

Comment (rushtong) [reply src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java]: Ah ... good catch, no, this is unintentional.

Comment (rushtong) [reply src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java]: There were several of these, all are updated now.

**Review by connorlbark — APPROVED** [review]
editor uses new config, tests run locally, code updates look good 👍

**Review by shaemarks — APPROVED** [review]
Updates to tests look consistent, and tests pass locally

**Pull request merged**